### PR TITLE
add trending to pinball

### PR DESCRIPTION
**Pull request by daverhub**

### CODE REVIEW

File: lib/pinballs.js [hunk 1]
```diff
@@ -52,6 +54,16 @@ class Pinballs {
         NLongitude        : oSetting.NLongitude,
         NBucketThreshold  : oSetting.NBucketThreshold
       });
+      this.trs[cityCode] = new Trending({
+        cityCode          : cityCode,
```
Comment (daverhub) [inline lib/pinballs.js:58]: add trending set to pinballs

File: lib/trending.js [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+
+    // pinball now has a rw cacheDB
+    this.cacheDB       = new Cache({ sCacheUrl : sCacheUrl });
+
+    const t0 = Date.now();
+    let t1;
+
+    const scanPattern = `tr:${cityCode}:*`; // used for keys
+    const setKey      = cache.getSortedSetName(cityCode);
+    console.log({ action: sAction, cityCode: cityCode, setKey: setKey });
```
Comment (daverhub) [inline lib/trending.js]: return `pb:${cityCode}:sortedSet`

File: lib/trending.js [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+    const cityCode     = this.cityCode;
+
+    // pinball now has a rw cacheDB
+    this.cacheDB       = new Cache({ sCacheUrl : sCacheUrl });
+
+    const t0 = Date.now();
+    let t1;
+
+    const scanPattern = `tr:${cityCode}:*`; // used for keys
+    const setKey      = cache.getSortedSetName(cityCode);
```
Comment (daverhub) [inline lib/trending.js]: returns `pb:${cityCode}:sortedSet`

**Comment by victusfate** [comment]
@daverhub just a note as soon as anything is merged into master it can be deployed on production. Best to keep in a branch until it's safe/ready for prod deployment (there are no additional branches here for dev etc)

What we can do is point sp0n-cache to your branch while you dev, and then when it's ready merge pinball into master and reference master again at sp0n-cache

**Pull request merged**